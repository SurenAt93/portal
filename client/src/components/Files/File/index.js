import React from 'react';

import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

// Icons
import FileIcon from '@material-ui/icons/Notes';

const File = ({ name }) => (
  <ListItem className="files__item">
    <div className="files__summary">
      <ListItemIcon><FileIcon /></ListItemIcon>
      <ListItemText primary={name}/>
    </div>
  </ListItem>
);

export default File;
