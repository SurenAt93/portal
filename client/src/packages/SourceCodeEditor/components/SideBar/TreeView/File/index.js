import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

import classNames from 'classnames';

// Icons
import FileIcon from '@material-ui/icons/Notes';

const File = ({ name, handleFileOpen, onContextMenu, open, path }) => (
  <ListItem className="files__item file">
    <div className="files__summary">
      <ListItemIcon><FileIcon /></ListItemIcon>
      <Button
        onContextMenu={onContextMenu}
        onClick={_ => handleFileOpen(path)}
        className={classNames({ 'files__file--open': open })}
      >
        <ListItemText primary={name}/>
      </Button>
    </div>
  </ListItem>
);

File.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  open: PropTypes.bool,
  handleFileOpen: PropTypes.func,
  onContextMenu: PropTypes.func.isRequired,
};

File.defaultProps = {
  handleFileOpen: _ => {},
  open: false,
};

export default File;
