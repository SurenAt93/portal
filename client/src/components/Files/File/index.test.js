import React from 'react';
import File from '.';

import Button from '@material-ui/core/Button';

describe('<File />', () => {
  it('should render normaly', () => {
    const component = shallow(<File name="app.js" />);

    expect(component).toMatchSnapshot();
  });

  it('button click should be call the handler function', () => {
    const clickFn = jest.fn();
    const component = shallow(<File name="app.js" handleFileOpen={clickFn} />);

    component
      .find(Button)
      .simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });
});
