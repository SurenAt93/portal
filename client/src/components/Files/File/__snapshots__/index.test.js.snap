// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<File /> should render normaly 1`] = `
<WithStyles(ListItem)
  className="files__item"
>
  <div
    className="files__summary"
  >
    <WithStyles(ListItemIcon)>
      <pure(Notes) />
    </WithStyles(ListItemIcon)>
    <WithStyles(Button)
      onClick={[Function]}
    >
      <WithStyles(ListItemText)
        primary="app.js"
      />
    </WithStyles(Button)>
  </div>
</WithStyles(ListItem)>
`;
